<h3 style="color: #001a79;">Exercise 3</h3>

<hr style="border-top: 1px solid #001a79;" />

It is somewhat interesting that `(5 * 4 * 3 * 2 * 1)` perfectly divides `(10 * 9 * 8 * 7 * 6)` - there's no remainder.

If we only wanted exactly four heads as opposed to five, the equivalent calculation would be `(10 * 9 * 8 * 7) / (4 * 3 * 2 * 1)`.

Does that evenly divide too? What is the formula in general?

Does it always come out as a positive whole number?

In [61]:
numerator = (10 * 9 * 8 * 7)
numerator

5040

In [62]:
denominator = (4 * 3 * 2 * 1)
denominator

24

In [64]:
numerator // denominator

210

This divides evenly. This is always the case because given the factorials on the numerator and denominator you will always have a factor of the numerator in the denominator. 

The general formula is: $${n \choose r} = {n! \over r!(n-r)!}$$

where n is the number of things to choose from, and we choose r of them. 

https://www.mathsisfun.com/combinatorics/combinations-permutations.html

If we use the above equation to solve $${10 \choose 4} = {10! \over 4!(10-4)!} = {10! \over (4!)(6!)} =$$

$${(10\times 9\times 8\times 7\times 6\times 5\times 4\times 3\times 2\times 1)} \over {(4\times 3\times 2\times 1)(6\times 5\times 4\times 3\times 2\times 1)}$$

$${(10\times 9\times 8\times 7)} \over {(4\times 3\times 2\times 1)}$$

As above we see we get a total of 210 combinations choosing 4 from 10. 

In [65]:
import math
numerator = (10 * 9 * 8 * 7)
denominator = (4 * 3 * 2 * 1)
numerator // denominator

210